modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosInstrumentActions.scala

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2016-2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.server.gmos
+
+import cats._
+import cats.implicits._
+import cats.data.NonEmptyList
+import edu.gemini.spModel.config2.Config
+import fs2.Stream
+import io.chrisdavenport.log4cats.Logger
+import seqexec.model.ActionType
+import seqexec.model.dhs.ImageFileId
+import seqexec.model.enum.NodAndShuffleStage._
+import seqexec.model.enum.ObserveCommandResult
+import seqexec.engine.Action
+import seqexec.engine.ParallelActions
+import seqexec.engine.Result
+import seqexec.server.Response
+import seqexec.server.SeqTranslate._
+import seqexec.server.StepType
+import seqexec.server.FileIdAllocated
+import seqexec.server.FileIdProvider
+import seqexec.server.InstrumentActions
+import seqexec.server.ObserveEnvironment
+import seqexec.server.ObserveActions
+import seqexec.server.ObserveContext
+import seqexec.server.SeqexecFailure
+import seqexec.server.ObserveActions._
+import seqexec.server.gmos.GmosController.Config._
+
+/**
+  * Gmos needs different actions for N&S
+  */
+class GmosInstrumentActions[F[_]: MonadError[?[_], Throwable]: Logger, A <: GmosController.SiteDependentTypes](
+  inst:   Gmos[F, A],
+  config: Config
+) extends InstrumentActions[F] {
+  override def observationProgressStream(
+    env: ObserveEnvironment[F]
+  ): Stream[F, Result[F]] =
+    ObserveActions.observationProgressStream(env)
+
+  // This tail is based on ObserveActions.observeTail
+  // But it can understand how to process Partial observations
+  // And can eventually return more than one result
+  private def observeTail(
+    fileId: ImageFileId,
+    dataId: String,
+    env:    ObserveEnvironment[F]
+  )(r:      ObserveCommandResult): F[Stream[F, Result[F]]] =
```

Comment:
Similar to the comment about `IO[Stream[IO` above. Are you sure you don't want to flatten to `Stream[F`?